lib/ASTGen/Sources/ASTGen/Macros.swift

```diff
@@ -60,14 +60,15 @@ enum MacroPluginKind: UInt8 {
 extension MacroRole {
   init(rawMacroRole: UInt8) {
```

Comment:
Am I missing why `swift_ASTGen_expandFreestandingMacro` and `swift_ASTGen_expandAttachedMacro` can't just be updated to take a `UInt32` instead?

Reply:
They could but it's not necessary; `swift::MacroRole` can only represent 32 distinct roles (because we use it for bitmasking where one bit corresponds to one role)

Reply:
Sure, it would just avoid the mapping is all 😅

Reply:
We already do the exact same mapping for `serialization::MacroRole`. We can probably just have one `RawMacroRole` that does this mapping between `uint32_t` and `uint8_t` values for macro roles.